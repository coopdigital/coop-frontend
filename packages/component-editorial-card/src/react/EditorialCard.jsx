import React from "react";
import PropTypes from "prop-types";
// import '../css/EditorialCard.pcss';

const EditorialCard = ({ title, text, link, type, testId }) => {
  const tagAttributes = {
    className: "coop coop-c-editorialcard coop-c-editorialcard--horizontal",
    role: type === "warn" || type === "error" ? "alert" : "status",
  };

  return (
    <div
      data-testid={testId}
      className={tagAttributes.className}
      role={tagAttributes.role}
    >
      <div className="coop-c-editorialcard__inner ">
        <h3 className="coop-c-editorialcard__title coop-c-notification__heading">{title}</h3>
        <p className="coop-c-notification__p">
          {link ? (
            <a href={link} className="coop-c-notification__link">
              {text}
            </a>
          ) : (
            text
          )}
        </p>
      </div>
    </div>
  );
};

EditorialCard.defaultProps = {
  type: "info",
};

EditorialCard.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
  type: PropTypes.oneOf(["info", "warn", "error", "success"]),
  testId: PropTypes.string,
};

export default EditorialCard;
