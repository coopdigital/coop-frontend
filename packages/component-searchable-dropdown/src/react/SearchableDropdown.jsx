import React, { useRef, useEffect } from "react";
import SearchableDropdownLibrary from "../lib/SearchableDropdownLibrary.js";

const SearchableDropdown = () => {
  const comboboxRef = useRef();

  useEffect(() => {
    new SearchableDropdownLibrary(comboboxRef.current).setup();
  }, []);

  return (
    <div ref={comboboxRef}>
      <div data-combobox>
        <label htmlFor="combobox">How you know them</label>
        <input
          name="combobox"
          className="coop-form__input"
          type="text"
          placeholder="Select option"
          data-input
          data-testid="combobox-input"
          role="combobox"
          aria-controls="comboboxvalues"
          aria-autocomplete="list"
          aria-expanded="false"
        />
        <div
          style={{ display: "none" }}
          data-dropdown
          data-testid="combobox-dropdown"
        >
          <ul id="comboboxvalues" role="listbox" data-list>
            {/* These valuse would be mapped from a data prop */}
            <li role="option" aria-selected="false" data-option>
              Partner
            </li>
            <li role="option" aria-selected="false" data-option>
              Spouse
            </li>
            <li role="option" aria-selected="false" data-option>
              Parent
            </li>
            <li role="option" aria-selected="false" data-option>
              Child
            </li>
            <li role="option" aria-selected="false" data-option>
              Brother
            </li>
            <li role="option" aria-selected="false" data-option>
              Grandchild
            </li>
            <li role="option" aria-selected="false" data-option>
              Father
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default SearchableDropdown;
