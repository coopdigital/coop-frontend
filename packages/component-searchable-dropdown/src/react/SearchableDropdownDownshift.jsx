/* eslint-disable indent */
import React, { useState } from 'react';
import { useCombobox } from 'downshift';
import PropTypes from 'prop-types';
import { matchSorter } from 'match-sorter';

const SearchableDropdown = ({
  className,
  compact,
  id,
  label,
  noResults,
  onSelect,
  options,
  placeholder,
  style,
}) => {
  const [inputItems, setInputItems] = useState(options);
  const sizeVariant = compact ? 'compact' : '';
  const noResultsText = noResults || 'No relevant options';
  const placeholderValue = placeholder || null;

  const {
    isOpen,
    closeMenu,
    getToggleButtonProps,
    getLabelProps,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    openMenu,
    getItemProps,
  } = useCombobox({
    id,
    items: inputItems,
    onInputValueChange: ({ inputValue }) => {
      setInputItems(matchSorter(options, inputValue));
    },
    onSelectedItemChange: ({ selectedItem: selected }) => {
      if (typeof onSelect === 'function') onSelect(selected);
      closeMenu();
    },
  });

  return (
    <div className={`coop-c-combobox ${sizeVariant} ${className}`} style={style}>
      <label {...getLabelProps()} htmlFor={`${id}-input`}>
        {label}
      </label>
      <div {...getComboboxProps()}>
        <input {...getInputProps({ onFocus: openMenu })} placeholder={placeholderValue} />
        <button type="button" {...getToggleButtonProps()} aria-label="toggle menu">
          <svg xmlns="http://www.w3.org/2000/svg" height="24" width="24">
            <path d="m12 15.375-6-6 1.4-1.4 4.6 4.6 4.6-4.6 1.4 1.4Z" />
          </svg>
        </button>
      </div>
      <ul {...getMenuProps()}>
        {isOpen && inputItems.length
          ? inputItems.map((item, index) => (
              <li {...getItemProps({ item, index })} key={`${item}${index}`}>
                {item}
              </li>
            ))
          : null}
        {isOpen && !inputItems.length ? (
          <li className="coop-c-combobox--null">{noResultsText}</li>
        ) : null}
      </ul>
    </div>
  );
};

export { SearchableDropdown };

SearchableDropdown.propTypes = {
  className: PropTypes.string,
  compact: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  noResults: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  placeholder: PropTypes.string,
  style: PropTypes.object,
};
