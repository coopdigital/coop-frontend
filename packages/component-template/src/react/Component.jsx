import React from "react";

const Component = () => <div class="coop-c-component">Example HTML</div>;

export default Component;
