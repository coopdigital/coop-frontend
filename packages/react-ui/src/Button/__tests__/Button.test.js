/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from "../src/index";

afterEach(cleanup);

describe("Button", () => {
  it("should render correctly", () => {
    const {getByText} = render(<Button>Button rendered</Button>);
    expect(getByText('Button rendered')).toBeInTheDocument();
  });

  it("should support all variants", () => {
    const expectedButtonText = ['Default', 'Primary', 'White', 'Grey'];
    const Variants = () => (
      <div>
        <Button>Default</Button>
        <Button variant="primary">Primary</Button>
        <Button variant="white">White</Button>
        <Button variant="grey">Grey</Button>
      </div>
    );
    const { getByText, asFragment } = render(<Variants />);

    expectedButtonText.forEach( expectedText => {
      expect(getByText(expectedText)).toBeInTheDocument();
    });

    const buttonVariantBlock = asFragment();
    expect(buttonVariantBlock).toMatchSnapshot();
  });

  it("should support all sizes", () => {
    const expectedButtonText = ['Small', 'Default/Medium'];

    const Sizes = () => (
      <div>
        <Button size="small">Small</Button>
        <Button size="medium">Default/Medium</Button>
      </div>
    );

    const {getByText, asFragment } = render(<Sizes />);

    expectedButtonText.forEach( expectedText => {
      expect(getByText(expectedText)).toBeInTheDocument();
    });

    const buttonSizesBlock = asFragment()
    expect(buttonSizesBlock).toMatchSnapshot();
  });

  it("should render different text", () => {
    const { rerender, getByText } = render(<Button>button</Button>);
    expect(getByText('button')).toBeInTheDocument();


    rerender(<Button>New text</Button>);
    expect(getByText('New text')).toBeInTheDocument();
  });

  it('should support full width buttons', () => {
    const {container, asFragment} = render(<Button fullWidth>Full width button</Button>);
    const fullwidthButton = asFragment();

    expect(container.firstChild).toHaveClass('coop-btn--full-width')
    expect(fullwidthButton).toMatchSnapshot();
  });

  it('should support loading state button prop', () => {
    const {container, asFragment} = render(<Button isLoading>Button is loading</Button>);
    const loadingButton = asFragment();

    expect(container.firstChild).toHaveClass('coop-btn--loading');
    expect(loadingButton).toMatchSnapshot();
  })


  it("should render empty button correctly", () => {
    const {container, asFragment} = render(<Button />);
    const emptyButtonBlock = asFragment();

    expect(emptyButtonBlock).toMatchSnapshot();
  });
});
