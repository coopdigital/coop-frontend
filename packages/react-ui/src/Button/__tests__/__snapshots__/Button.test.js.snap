// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render empty button correctly 1`] = `
<DocumentFragment>
  <button
    class="coop-btn"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button should support all sizes 1`] = `
<DocumentFragment>
  <div>
    <button
      class="coop-btn coop-btn--small"
      data-testid="button-small"
      type="button"
    >
      Small
    </button>
    <button
      class="coop-btn"
      type="button"
    >
      Default/Medium
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button should support all variants 1`] = `
<DocumentFragment>
  <div>
    <button
      class="coop-btn"
      type="button"
    >
      Default
    </button>
    <button
      class="coop-btn coop-btn--primary"
      data-testid="button-primary"
      type="button"
    >
      Primary
    </button>
    <button
      class="coop-btn coop-btn--white"
      data-testid="button-white"
      type="button"
    >
      White
    </button>
    <button
      class="coop-btn coop-btn--grey"
      data-testid="button-grey"
      type="button"
    >
      Grey
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button should support full width buttons 1`] = `
<DocumentFragment>
  <button
    class="coop-btn coop-btn--full-width"
    type="button"
  >
    Full width button
  </button>
</DocumentFragment>
`;

exports[`Button should support loading state button prop 1`] = `
<DocumentFragment>
  <button
    class="coop-btn coop-btn--loading"
    type="button"
  >
    Button is loading
  </button>
</DocumentFragment>
`;
